### ISSUE TITLE
Search results incomplete

### ISSUE THREAD

**Issue by sbulen** [SimpleMachines/SMF#5978]
#### Description
My search results are woefully incomplete in 2.1.  The indexing doesn't appear to get all the content. 
 This is true for adding new posts as well as after rebuilding indexes.  

I've seen this on windows & linux.  Using a custom index, fulltext index, and no index.

Note if you migrated a 2.0 board over, that fully migrated search index is complete as of migration & seems to work OK for searching migrated content.  I think there is a problem adding/using new posts in 2.1.  

**_EDIT:  It seems to work much better if you check the box for "show results as messages".  Maybe it's not that the index is incomplete, but in the display somewhere...  Or both indexing & display._**  

### Steps to reproduce
1. Post 10 messages, then search for a word you know is in all 10.  
2. Alternate: Delete your search index & rebuild it.

### Environment (complete as necessary)
- Version/Git revision:  Current
- Database Type:  mysql
- Database Version:  5.7
- PHP Version:  7.3 & 5.4

### Additional information/references

**Comment by sbulen** [comment]
Here is an example.  I posted this 5 times:
![image](https://user-images.githubusercontent.com/23568484/73391638-2feb6780-428d-11ea-819e-9b425892e7dc.png)

This is a search without the 'Show results as messages' option checked:
![image](https://user-images.githubusercontent.com/23568484/73391705-4db8cc80-428d-11ea-8f52-908cb8c124ca.png)

This is a search with the 'Show results as messages' option checked: 
![image](https://user-images.githubusercontent.com/23568484/73391754-64f7ba00-428d-11ea-8d66-8af64520d520.png)

**Comment by sbulen** [comment]
This is another example...  I have, oh, about 45 copies of this posted.  This is a partial image:
![image](https://user-images.githubusercontent.com/23568484/73393755-1815e280-4291-11ea-8908-940627cf2bc9.png)

Yet, if I search for the word Poetry, I get:
![image](https://user-images.githubusercontent.com/23568484/73393816-324fc080-4291-11ea-9596-e2b0998debbc.png)

**Comment by albertlast** [comment]
What is yours index settings?
Did you check postgres too?
Did you clean the search cache?

**Comment by sbulen** [comment]
OK, I am going to close this out...  

There were two concerns above:
1) Multiple posts in the same topic get collapsed into one search result, when the "show results as messages" option is NOT checked.  They do appear as unique items when that option is checked.  This appears to be deliberate, and it behaves this way in 2.0.  It does tidy up the search results.  

Apparently a feature, not a bug.


2) Posts not showing up at all under certain circumstances.  I have only been able to reproduce this a couple of times (among many attempts...).  When I was able to reproduce, it was with multiple massive posts with many new words with multi-byte characters.  I believe the indexing can get backlogged somehow, with no errors reported in apache or smf.  

Very rare & I cannot reproduce consistently.  

And...  _It corrects itself after a while..._  So at worst a delay, not lost data.
